jdaviz/configs/default/plugins/collapse/collapse.py

```diff
@@ -74,30 +62,28 @@ def _on_data_updated(self, msg):
 
     @observe('selected_data_item')
     def _on_data_item_selected(self, event):
-        self._selected_data = next((x for x in self.data_collection
-                                    if x.label == event['new']))
+        data_label = event['new']
+        if data_label not in self.data_collection.labels:
+            return
+        self._selected_data = self.data_collection[self.data_collection.labels.index(data_label)]
+        self._selected_cube = self._selected_data.get_object(cls=Spectrum1D, statistic=None)
+        self.spectral_unit = self._selected_cube.spectral_axis.unit.to_string()
 
         # Also set the spectral min and max to default to the full range
-        cube = self._selected_data.get_object(cls=SpectralCube)
-        self.selected_subset = "None"
-        self.spectral_min = cube.spectral_axis[0].value
-        self.spectral_max = cube.spectral_axis[-1].value
-        self.spectral_unit = str(cube.spectral_axis.unit)
-
-        self.axes = list(range(len(self._selected_data.shape)))
+        self.selected_subset = "None"  # This calls self._on_subset_selected()
 
     @observe("selected_subset")
     def _on_subset_selected(self, event):
+        if self._selected_data is None:
+            return
+
         # If "None" selected, reset based on bounds of selected data
         if self.selected_subset == "None":
```

Comment:
Might be out-of-scope, but we've updated some of the plugins to use "Entire Spectrum" in place of "None".  We probably should just go through the remaining plugins and get them all to be consistent at some point soon.

Reply:
I can change it here if there is consensus. If you do a code suggestion, you will get credit for it. If not, I'll change it at the next commit when I get a chance. Up to you.

Reply:
There are too many places to change it for a code suggestion, but I can add a commit if you'd prefer (but also don't really care about the credit if you want to just handle it in the same commit as https://github.com/spacetelescope/jdaviz/pull/1006#discussion_r803118536)

Reply:
Oh, I mean just to change it for this plugin in this PR. But if you want to do it for the whole app, I agree that is a separate PR.